common/src/main/java/org/apache/celeborn/common/network/util/NettyUtils.java

```diff
@@ -106,17 +106,32 @@ public static String getRemoteAddress(Channel channel) {
    * released by the executor thread rather than the event loop thread. Those thread-local caches
    * actually delay the recycling of buffers, leading to larger memory usage.
    */
+  @SuppressWarnings("deprecation")
   public static PooledByteBufAllocator createPooledByteBufAllocator(
       boolean allowDirectBufs, boolean allowCache, int numCores) {
     if (numCores == 0) {
       numCores = Runtime.getRuntime().availableProcessors();
     }
+    // After upgrade to Netty 4.1.75, there are 2 behavior changes of this method:
```

Comment:
is this related to the perf gain you mentioned? @FMX

Reply:
Yes, and there are some other reasons to perf gain.